fairlearn/postprocessing/_plotting.py

```diff
@@ -104,3 +107,286 @@ def plot_threshold_optimizer(threshold_optimizer, ax=None, show_plot=True):
+
+    specified_sf_to_plot : list or None, default=None
+        If None, plot information about all known sensitive features.
+        If list, plot information about sensitive features in list
+    """
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise RuntimeError(_MATPLOTLIB_IMPORT_ERROR_MESSAGE)
+
+    sensitive_features, all_sf, samples_each_sf, n_test_samples = \
+        _check_sensitive_features(sensitive_features)
+
+    sf_to_plot = specified_sf_to_plot if specified_sf_to_plot else all_sf
+
+    # create dict to keep track of proba distribution for each sf, by creating array of the form
+    # [#of probas in interval 0 - .1, ... , #of probas in interval .9 - 1]
+    statistics_dict = _create_statistics_dict(sf_to_plot, np.zeros(10))
+
+    # loop through A and Y_pred_proba simultaneously, and only once
+    for i in range(n_test_samples):
+        sf = sensitive_features.iloc[i]
+        if sf in statistics_dict:
+            stats_array = statistics_dict[sf]
+            # append predict proba to correct interval
+            stats_array[math.floor(Y_pred_proba[i]*10)] += 1
+
+    # create arrays used for making the plot
+    stats_arrays = statistics_dict.values()
+    nmbr_sf_to_plot = len(sf_to_plot)
+    bottom = np.zeros(nmbr_sf_to_plot)
+    colorbrewer = ['#a6cee3', '#1f78b4', '#b2df8a', '#33a02c',
+                   '#fdbf6f', '#ff7f00', '#e31a1c', '#fb9a99', '#6a3d9a', '#cab2d6']
+    labels = []
+    handles = []
+
+    # Start the making plot
+    plt.figure(figsize=(nmbr_sf_to_plot, 6))
+    for i in range(10):
+
+        # Calculate percentages of probas in each interval for each group
+        nmbr_of_probas_in_interval = np.array([stats_array[i] for stats_array in stats_arrays])
+        nmbr_of_each_group = np.array([samples_each_sf[sf] for sf in sf_to_plot])
+        nmbr_probas_percentages = nmbr_of_probas_in_interval/nmbr_of_each_group*100
+
+        # Plot next bar
+        handle = plt.bar(sf_to_plot, nmbr_probas_percentages,
+                         bottom=bottom, color=colorbrewer[i], alpha=.8)
+
+        # Keep track of handles and labels for the legend
+        handles.insert(0, handle)
+        labels.insert(0, '{} - {}'.format(i/10, (i+1)/10))
+
+        # Create text annotation
+        for xpos, ypos, yval in zip(sf_to_plot, bottom+nmbr_probas_percentages/2,
+                                    nmbr_probas_percentages):
+            if yval >= 5:
+                plt.text(xpos, ypos, '{}%'.format(round(yval, 1)), ha="center", va="center")
+
+        # Update y-positions for next bars
+        bottom += nmbr_probas_percentages
+
+    plt.ylabel('distribution of predict_proba')
+    plt.xlabel('Sensitive feature')
+    plt.xticks(rotation=45, ha='right')
+    plt.legend(labels=labels, handles=handles, bbox_to_anchor=(1, .9))
+    plt.show()
+
+
+def plot_positive_predictions(sensitive_features, Y_pred, specified_sf_to_plot=None):
```

Comment:
For `plot_positive_predictions`, I think it is possible to create a plot using `by_group.plot()` that gives the same information, but I think it will be hard to have it include the percentages in the bars without writing code for that (which would end up being more or less the same as in the function written now). So I guess the question is: do we want it to have the percentages included or not. Personally, I feel like the plots produced by the functions in _plotting.py are more intuitive than using `by_group.plot()`, but then again I’m obviously rather biased as I wrote the functions ;). Wdyt @romanlutz @hildeweerts

Reply:
Can you attach a picture? It's a bit hard to imagine 🙂